### PR TITLE
chore: Replace 'M1' with 'Apple Silicon' in the doc

### PR DESCRIPTION
**Pull request by ken0nek**
## Summary:

Use Apple Silicon instead of M1 because we have M2 and other variations too.

I did not change the name of `__apply_Xcode_12_5_M1_post_install_workaround` as it's in use.

## Changelog:

[GENERAL] [FIXED] - Use Apple Silicon instead of M1

## Test Plan:

None as no behavioral changes

### CODE REVIEW

File: packages/react-native/template/android/build.gradle [hunk 1]
```diff
@@ -7,7 +7,7 @@ buildscript {
         compileSdkVersion = 33
         targetSdkVersion = 33
 
-        // We use NDK 23 which has both M1 support and is the side-by-side NDK version from AGP.
+        // We use NDK 23 which has both Apple Silicon support and is the side-by-side NDK version from AGP.
```
Comment (cortinico) [inline packages/react-native/template/android/build.gradle:10]: Can you undo this change?
This is our template and it will create more noise than anything

Comment (ken0nek) [reply packages/react-native/template/android/build.gradle]: Thank you for the review! Addressed with [955f8de](https://github.com/facebook/react-native/pull/38344/commits/955f8dea1f4705aa9ffdd10105c18bb55b7a8ec0)
